--- mariadb-10.1.orig/support-files/mariadb@.service.in
+++ mariadb-10.1/support-files/mariadb@.service.in
@@ -65,6 +65,8 @@ ProtectHome=true
 # Execute pre and post scripts as root, otherwise it does it as User=
 PermissionsStartOnly=true
 
+@SYSTEMD_EXECSTARTPRE@
+
 # Perform automatic wsrep recovery. When server is started without wsrep,
 # galera_recovery simply returns an empty string. In any case, however,
 # the script is not expected to return with a non-zero status.
