From: Debian MySQL Maintainers <pkg-mysql-maint@lists.alioth.debian.org>
Date: Thu, 10 Aug 2017 20:40:30 +0200
Subject: Add_default_ExecStartPre_to_mariadb@.service

---
 support-files/mariadb@.service.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/support-files/mariadb@.service.in b/support-files/mariadb@.service.in
index 02be006..183b68b 100644
--- a/support-files/mariadb@.service.in
+++ b/support-files/mariadb@.service.in
@@ -67,6 +67,8 @@ ProtectHome=true
 # Execute pre and post scripts as root, otherwise it does it as User=
 PermissionsStartOnly=true
 
+@SYSTEMD_EXECSTARTPRE@
+
 # Perform automatic wsrep recovery. When server is started without wsrep,
 # galera_recovery simply returns an empty string. In any case, however,
 # the script is not expected to return with a non-zero status.
