From: Debian MySQL Maintainers <pkg-mysql-maint@lists.alioth.debian.org>
Date: Sun, 23 Apr 2017 15:10:44 +0200
Subject: extend_default_test_timeout_for_tokudb

---
 mysql-test/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mysql-test/CMakeLists.txt b/mysql-test/CMakeLists.txt
index 1c7cfaf..bc45177 100644
--- a/mysql-test/CMakeLists.txt
+++ b/mysql-test/CMakeLists.txt
@@ -64,7 +64,7 @@ ELSE()
 ENDIF()
 
 
-SET(MTR_FORCE perl ./mysql-test-run.pl --force)
+SET(MTR_FORCE perl ./mysql-test-run.pl --force --testcase-timeout=30 --suite-timeout=540)
 IF(EXISTS ${CMAKE_SOURCE_DIR}/mysql-test/suite/nist)
   SET(TEST_NIST ${MTR_FORCE} --comment=nist suite=nist ${EXP} && 
     ${MTR_FORCE} --comment=nist --force --suite=nist+ps ${EXP})
