Description: Fix spelling errors
Author: Otto Kekäläinen <otto@seravo.fi>
Last-Update: 2014-10-02

--- a/storage/connect/tabxml.cpp
+++ b/storage/connect/tabxml.cpp
@@ -1798,7 +1798,7 @@ void XMULCOL::ReadColumn(PGLOBAL g)
 
       if (N > Tdbp->Limit) {
         N = Tdbp->Limit;
-        sprintf(g->Message, "Mutiple values limited to %d", Tdbp->Limit);
+        sprintf(g->Message, "Multiple values limited to %d", Tdbp->Limit);
         PushWarning(g, Tdbp);
         } // endif N
 
--- a/man/mysqldump.1
+++ b/man/mysqldump.1
@@ -1275,7 +1275,7 @@ to point the slave to the correct master
 .\" max-allowed-packet option: mysqldump
 \fB\-\-max\-allowed\-packet=\fR\fB\fIlength\fR\fR
 .sp
-Sets the maximum packet length to send to or recieve from server\&.
+Sets the maximum packet length to send to or receive from server\&.
 .RE
 .sp
 .RS 4
--- a/storage/connect/mysql-test/connect/r/xml_mult.result
+++ b/storage/connect/mysql-test/connect/r/xml_mult.result
@@ -51,14 +51,14 @@ WEB	XQuery Kick Start	en	Per Bothner	200
 WEB	XQuery Kick Start	en	Kurt Cagle	2003	49.99
 WEB	Learning XML	en	Erik T. Ray	2003	39.95
 Warnings:
-Warning	1105	Mutiple values limited to 3
+Warning	1105	Multiple values limited to 3
 # One line lost because the where clause is applied only on the first 3 rows
 SELECT category, title, author, price FROM bookstore WHERE author LIKE 'J%';
 category	title	author	price
 CHILDREN	Harry Potter	J K. Rowling	29.99
 WEB	XQuery Kick Start	James McGovern	49.99
 Warnings:
-Warning	1105	Mutiple values limited to 3
+Warning	1105	Multiple values limited to 3
 #
 # Testing concatenated values
 #
@@ -91,12 +91,12 @@ CHILDREN	Harry Potter	en	J K. Rowling	20
 WEB	XQuery Kick Start	en	James McGovern, Per Bothner, Kurt Cagle, James Linn	2003	49.99
 WEB	Learning XML	en	Erik T. Ray	2003	39.95
 Warnings:
-Warning	1105	Mutiple values limited to 4
+Warning	1105	Multiple values limited to 4
 # The where clause is applied on the concatenated column result
 SELECT category, title, author, price FROM bookstore WHERE author LIKE 'J%';
 category	title	author	price
 CHILDREN	Harry Potter	J K. Rowling	29.99
 WEB	XQuery Kick Start	James McGovern, Per Bothner, Kurt Cagle, James Linn	49.99
 Warnings:
-Warning	1105	Mutiple values limited to 4
+Warning	1105	Multiple values limited to 4
 DROP TABLE bookstore;
