From: Debian MySQL Maintainers <pkg-mysql-maint@lists.alioth.debian.org>
Date: Sun, 23 Apr 2017 15:10:44 +0200
Subject: mips-machine

---
 cmake/package_name.cmake | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cmake/package_name.cmake b/cmake/package_name.cmake
index 4930a6b..7681f78 100644
--- a/cmake/package_name.cmake
+++ b/cmake/package_name.cmake
@@ -34,6 +34,10 @@ IF(NOT VERSION)
       SET(DEFAULT_MACHINE "mips")
     ENDIF()
 
+    IF(NOT 64BIT AND CMAKE_SYSTEM_PROCESSOR MATCHES "^mips64")
+      SET(DEFAULT_MACHINE "mips")
+    ENDIF()
+
     IF(CMAKE_SYSTEM_NAME MATCHES "Windows")
       SET(NEED_DASH_BETWEEN_PLATFORM_AND_MACHINE 0)
       SET(DEFAULT_PLATFORM "win")
