Description: Provides LSB headers for example initscript (Closes: #778762)
Author: Jan Wagner <waja@cyconet.org>
--- a/support-files/mysqld_multi.server.sh
+++ b/support-files/mysqld_multi.server.sh
@@ -14,8 +14,24 @@
 # Version 1.0
 #
 
-basedir=/usr/local/mysql
-bindir=/usr/local/mysql/bin
+### BEGIN INIT INFO
+# Provides:          mysqld_multi
+# Required-Start:    $remote_fs $syslog
+# Required-Stop:     $remote_fs $syslog
+# Should-Start:      $network $named $time
+# Should-Stop:       $network $named $time
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: Start and stop multiple mysql database server daemon instances
+# Description:       Controls multiple MariaDB database server daemon instances
+### END INIT INFO
+
+PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
+NAME=mysqld_multi
+DESC=mysqld_multi
+
+basedir=/usr
+bindir=/usr/bin
 
 if test -x $bindir/mysqld_multi
 then
