Author: Otto Kekäläinen <otto@seravo.fi>
Subject: Update test main.sll to match backported new cacert.pem
Last-Update: 2015-01-29

diff --git a/mysql-test/r/ssl.result b/mysql-test/r/ssl.result
--- a/mysql-test/r/ssl.result
+++ b/mysql-test/r/ssl.result
@@ -3,10 +3,10 @@ Variable_name	Value
 Ssl_cipher	DHE-RSA-AES256-SHA
 SHOW STATUS LIKE 'Ssl_server_not_before';
 Variable_name	Value
-Ssl_server_not_before	Feb 20 02:55:06 2010 GMT
+Ssl_server_not_before	Jan 29 13:33:36 2015 GMT
 SHOW STATUS LIKE 'Ssl_server_not_after';
 Variable_name	Value
-Ssl_server_not_after	Sep  3 02:55:06 2030 GMT
+Ssl_server_not_after	Jan 24 13:33:36 2035 GMT
 drop table if exists t1,t2,t3,t4;
 CREATE TABLE t1 (
 Period smallint(4) unsigned zerofill DEFAULT '0000' NOT NULL,
