Author: Otto Kekäläinen <otto@seravo.fi>
Description: Remove upstream debian/* handling that is obsolete

Index: gitwd/CMakeLists.txt
===================================================================
--- gitwd.orig/CMakeLists.txt
+++ gitwd/CMakeLists.txt
@@ -460,12 +460,6 @@ CONFIGURE_FILE(
     ${CMAKE_SOURCE_DIR}/cmake/info_macros.cmake.in
     ${CMAKE_BINARY_DIR}/info_macros.cmake @ONLY)
 
-IF(DEB)
-  CONFIGURE_FILE(
-    ${CMAKE_SOURCE_DIR}/debian/mariadb-server-10.0.files.in
-    ${CMAKE_SOURCE_DIR}/debian/mariadb-server-10.0.files)
-ENDIF(DEB)
-
 # Handle the "INFO_*" files.
 INCLUDE(${CMAKE_BINARY_DIR}/info_macros.cmake)
 # Source: This can be done during the cmake phase, all information is
